database/mongoDb.js

```diff
@@ -0,0 +1,214 @@
+            type:house.type,
+            year_built: house.year_built,
+            heating: house.heating,
+            cooling: house.cooling,
+            parking: house.parking,
+            lot: house.lot,
+            days_on_zillow: house.days_on_zillow,
+            bedrooms: house.bedrooms,
+            bathrooms: house.bathrooms,
+            interiorheating: house.int_heating,
+            interiorcooling: house.int_cooling,
+            appliances: house.appliances,
+            kitchen: house.kitchen,
+            flooring: house.flooring,
+            sqft: house.sqft
+          }
+
+        }
+      });
+
+    });
+
+    return params;
+  }
+
+
+  //function to load feature mock data
+  const loadFeatures = (callback, numRecords = 100, params = []) => {
+    console.log(documentCount);
+    if (params.length === 0) {
+      features.createFeatures(numRecords).forEach(house => {
+        params.push({
+          insertOne : {
+            document: {
+              house_id: ++documentCount,
+              type:house.type,
+              year_built: house.year_built,
+              heating: house.heating,
+              cooling: house.cooling,
+              parking: house.parking,
+              lot: house.lot,
+              days_on_zillow: house.days_on_zillow,
+              bedrooms: house.bedrooms,
+              bathrooms: house.bathrooms,
+              interiorheating: house.int_heating,
+              interiorcooling: house.int_cooling,
+              appliances: house.appliances,
+              kitchen: house.kitchen,
+              flooring: house.flooring,
+              sqft: house.sqft
+            }
+          }
+        });
+      });
+    } else {
+      params.forEach(function (value, index, array) {
+        value.insertOne.document.house_id = ++documentCount;
+      });
+    }
+
+    featuresTable.bulkWrite(params).then(()=> {
+      console.log(`Loaded ${numRecords}`);
+      callback();
+    }).catch((err)=> {
+      console.log('an error occured');
+      console.log(err);
+    });
+
+    };
+
+  /////////////////////////Retreiving Facts and Features///////////////////////////
+  const getFeatures = (id, callback) => {
+    console.log(id);
+    featuresTable.find({ house_id: id }, function (err, docs) {
+      if (err) {
+        console.log(err);
+        callback(err, null);
+      } else {
+        console.log(docs);
+        callback(null, docs);
+      }
+    });
+  };
+
+  const getBedBaths = (id, callback) => {
+    console.log(id);
+    featuresTable.find({ house_id: id }, function (err, docs) {
+      if (err) {
+        console.log(err);
+        callback(err, null);
+      } else {
+        console.log(docs);
+        callback(null, docs);
+      }
+    });
+    };
+
+    var getAllFeatures = (page, callback) => {
+
+    };
+
+    var postFeatures = (features, callback) => {
+      if (isNaN(id)) {
+        callback('no Id to insert', null);
+      }
+      features.house_id = undefined;

```

Comment:
undefined is a value set by the interpreter. Use null.

Reply:
If I do that won't mongoose try to insert null into house_id for this record and error out?  I used undefined to make the key effectively 'dissappear' at least that is what went through my mind.